compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java

```diff
@@ -476,7 +476,10 @@ private void defineFunction(DataInputStream dataInStream) throws IOException {
                     }
 
                     resourceFunction.pathSegmentSymbols = pathSegmentSymbols;
-                    objectTypeSymbol.attachedFuncs.add(resourceFunction);
+
+                    if (!isImmutable(this.currentStructure.flags)) {
```

Comment:
Do we know why we need to special case like this?

Reply:
May be related to https://github.com/ballerina-platform/ballerina-lang/blob/v2201.7.0/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java#L1712

Reply:
Yeah, In here https://github.com/ballerina-platform/ballerina-lang/blob/v2201.7.0/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java#L1712, here we are adding immutable functions first and we don't need another round to add the functions again. That's why I ignore adding immutable functions again

Reply:
So can't we get rid of that instead?

Reply:
This seems like a known [issue (#30051)](https://github.com/ballerina-platform/ballerina-lang/issues/30051) that needs a proper workaround separately. I have added a TODO comment

Reply:
But if the function is now available, maybe that was fixed somehow? Did you check if that is still needed?